src/agent/src/device.rs

```diff
@@ -444,6 +499,7 @@ async fn add_device(
 
     match device.field_type.as_str() {
         DRIVER_BLK_TYPE => virtio_blk_device_handler(device, spec, sandbox, devidx).await,
+        DRIVER_BLK_CCW_TYPE => virtio_blk_ccw_device_handler(device, spec, sandbox, devidx).await,
```

Comment:
@jodh-intel re https://github.com/kata-containers/kata-containers/pull/2027#discussion_r651665607:
I don't think I can limit the CCW implementation to s390x except by doing the following:
- somehow turn this `match` map into a mutable variable
- add a function to expand it, adding blk-ccw on s390x, doing nothing on !s390x
- `match` upon that map

Otherwise, this won't build on !s390x if the CCW implementation is limited to s390x.

I think that's a bit clumsy. WDYT?

Reply:
Couldn't you do something like this?:

```diff
diff --git a/src/agent/src/device.rs b/src/agent/src/device.rs
index 5af91c38..7177c4c3 100644
--- a/src/agent/src/device.rs
+++ b/src/agent/src/device.rs
@@ -394,6 +394,7 @@ async fn virtio_blk_device_handler(
 }

 // device.id should be a CCW path string
+#[cfg(target_arch = "s390x")]
 #[instrument]
 async fn virtio_blk_ccw_device_handler(
     device: &Device,
@@ -407,6 +408,17 @@ async fn virtio_blk_ccw_device_handler(
     update_spec_device_list(&dev, spec, devidx)
 }

+#[cfg(not(target_arch = "s390x"))]
+#[instrument]
+async fn virtio_blk_ccw_device_handler(
+    _device: &Device,
+    _spec: &mut Spec,
+    _sandbox: &Arc<Mutex<Sandbox>>,
+    _devidx: &DevIndex,
+) -> Result<()> {
+    Err(anyhow!("ccw only supported on s390x"))
+}
+
 // device.Id should be the SCSI address of the disk in the format "scsiID:lunID"
 #[instrument]
 async fn virtio_scsi_device_handler(
```

Reply:
Oh yes, that's a possibility. Let me get back :)